dali/operators/image/paste/multipaste.cc

```diff
@@ -0,0 +1,153 @@
+
+  auto& tp = ws.GetThreadPool();
+
+  auto batch_size = output.shape().num_samples();
+
+  TYPE_SWITCH(images.type().id(), type2id, InputType, (uint8_t, int16_t, int32_t, float), (
+      TYPE_SWITCH(output_type_, type2id, OutputType, (uint8_t, int16_t, int32_t, float), (
+          {
+            using Kernel = kernels::PasteCPU<OutputType, InputType>;
+            auto in_view = view<const InputType, 3>(images);
+            auto out_view = view<OutputType, 3>(output);
+            for (int i = 0; i < batch_size; i++) {
+              auto paste_count = in_idx_[i].shape[0];
+              memset(out_view[i].data, 0,
+                     out_view[i].shape[0] * out_view[i].shape[1] * out_view[i].shape[2]);
+
+              if (no_intersections_[i]) {
+                for (int iter = 0; iter < paste_count; iter++) {
+                  int from_sample = in_idx_[i].data[iter];
+                  int to_sample = i;
+
+                  tp.AddWork(
+                      [&, i, iter, from_sample, to_sample, in_view, out_view](int thread_id) {
+                        kernels::KernelContext ctx;
+                        auto tvin = in_view[from_sample];
+                        auto tvout = out_view[to_sample];
+
+                        auto in_anchor_view = GetInAnchors(i, iter);
+                        auto in_shape_view = GetShape(i, iter, Coords(
+                            raw_input_size_mem_.data() + 2 * from_sample, dali::TensorShape<>(2)));
+                        auto out_anchor_view = GetOutAnchors(i, iter);
+                        kernel_manager_.Run<Kernel>(thread_id, to_sample, ctx, tvout, tvin,
+                                        in_anchor_view, in_shape_view, out_anchor_view);
+                      },
+                      out_shape.tensor_size(to_sample));
+                }
+              } else {
+                tp.AddWork(
+                    [&, i, paste_count, in_view, out_view](int thread_id) {
+                      for (int iter = 0; iter < paste_count; iter++) {
+                        int from_sample = in_idx_[i].data[iter];
+                        int to_sample = i;
+
+                        kernels::KernelContext ctx;
+                        auto tvin = in_view[from_sample];
+                        auto tvout = out_view[to_sample];
+
+                        auto in_anchor_view = GetInAnchors(i, iter);
+                        auto in_shape_view = GetShape(i, iter, Coords(
+                            raw_input_size_mem_.data() + 2 * from_sample, dali::TensorShape<>(2)));
+                        auto out_anchor_view = GetOutAnchors(i, iter);
+                        kernel_manager_.Run<Kernel>(thread_id, to_sample, ctx, tvout, tvin,
+                                        in_anchor_view, in_shape_view, out_anchor_view);
+                      }
+                    },
+                    paste_count * out_shape.tensor_size(0));
+              }
+            }
+          }
```

Comment:
A nitpick or a suggestion - this block of code is _very_ large - handling build errors here must be a nightmare. You could extract this into a templated function - for example, just get out_view and in_view here and pass them into a function - it will infer the types from the arguments and you can get the kernel object there.

Reply:
I was about to suggest the same :)